src/components/authentication/sign-in-modal.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  Box,
+  Button,
+  Container,
+  Grid,
+  Link,
+  Modal,
+  Paper,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { useDispatch, useSelector } from "react-redux";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { useSnackbar } from "material-ui-snackbar-provider";
+
+import { AppDispatch } from "../../store";
+import { FormInput } from "../../models/form";
+import { userSignIn } from "../../store/authentication/user-action";
+import { UserState } from "../../models/user-state";
+
+const SignIn = ({ openSignInModal, setOpenSignInModal }) => {
+  const dispatch: AppDispatch = useDispatch();
+
+  const snackbar = useSnackbar();
+
+  const { success, error } = useSelector((state: UserState) => state.user);
+
+  const { register, handleSubmit } = useForm<FormInput>({
+    defaultValues: { email: "", password: "" },
+  });
+
+  const submitHandler = (data: FormInput) => {
+    dispatch(userSignIn(data));
+  };
+
+  useEffect(() => {
+    if (success) {
+      snackbar.showMessage("You have been signed in!");
+    }
+    if (error) {
+      snackbar.showMessage("Something went wrong!");
+    }
+  }, [success, error, snackbar]);
+
+  return (
+    <Modal
+      open={openSignInModal}
+      onClose={() => setOpenSignInModal(false)}
+      aria-labelledby="modal-modal-title"
+      aria-describedby="modal-modal-description"
+    >
+      <Container maxWidth="sm">
+        <Paper elevation={2}>
+          <Box padding={2}>
+            <Typography variant="h5" padding={2}>
+              Sign In
+            </Typography>
+            <form onSubmit={handleSubmit(submitHandler)}>
+              <Stack spacing={1} alignItems="baseline">
+                <Grid container direction="column" item xs={5}>
```

Comment:
This seems to be everywhere. But the Grid element should be either container or an item, not both